src/aind_data_transfer/jobs/zarr_upload_job.py

```diff
@@ -0,0 +1,294 @@
+from pathlib import Path
+from time import sleep
+from typing import Any, Optional, Tuple, List
+
+from aind_data_transfer.util import setup_logging
+from aind_data_transfer.config_loader.base_config import (
+    ConfigError,
+    BasicUploadJobConfigs,
+)
+from aind_data_transfer.jobs.basic_job import BasicJob
+from aind_data_transfer.transformations.converters import log_to_acq_json, acq_json_to_xml
+from aind_data_transfer.transformations.ome_zarr import write_files
+from aind_data_transfer.util.dask_utils import get_client, get_deployment
+from aind_data_transfer.util.env_utils import find_hdf5plugin_path
+from aind_data_transfer.util.file_utils import get_images
+from aind_data_transfer.util.s3_utils import upload_to_s3
+from aind_data_transfer.transformations.file_io import read_toml, read_imaging_log, write_acq_json, write_xml
+from aind_data_transfer.transformations.ng_link_creation import write_json_from_zarr
+
+import yaml
+from numcodecs import blosc
+from pydantic import Field, BaseSettings
+from aind_data_schema.data_description import Modality
+
+
+_CLIENT_CLOSE_TIMEOUT = 300  # seconds
+_CLIENT_SHUTDOWN_SLEEP_TIME = 30  # seconds
+
+
+class ZarrConversionConfigs(BaseSettings):
+    n_levels: Optional[int] = Field(
+        1, description="Number of levels to use for the pyramid. Default is 1."
+    )
+    scale_factor: Optional[int] = Field(
+        2, description="Scale factor to use for the pyramid. Default is 2."
+    )
+    chunk_shape: Optional[Tuple[int, int, int, int, int]] = Field(
+        (1, 1, 256, 256, 256),
+        description="5D Chunk shape to use for the zarr Array. Default is (1, 1, 256, 256, 256).",
+    )
+    voxel_size: Optional[Tuple[float, float, float]] = Field(
+        None,
+        description="Voxel size to use for the zarr Array. if None, will attempt to parse from the image metadata. Default is None.",
+    )
+    codec: Optional[str] = Field(
+        "zstd",
+        description="Blosc codec to use for compression. Default is zstd.",
+    )
+    clevel: Optional[int] = Field(
+        1, description="Blosc compression level to use. Default is 1."
+    )
+    do_bkg_subtraction: Optional[bool] = Field(
+        False,
+        description="Whether to subtract the background image from the raw data. Default is False.",
+    )
+    exclude_patterns: Optional[List[str]] = Field(
+        None,
+        description="List of patterns to exclude from the zarr conversion. Default is None.",
+    )
+    create_ng_link: Optional[bool] = Field(
+        False,
+        description="Whether to create a neuroglancer link. Default is False.",
+    )
+    ng_vmin: Optional[float] = Field(
+        0,
+        description="Default minimum of the neuroglancer display range. Default is 0."
+    )
+    ng_vmax: Optional[float] = Field(
+        200.0,
+        description="Default maximum of the neuroglancer display range. Default is 200."
+    )
+
+    @classmethod
+    def from_yaml(cls, yaml_path: Path):
```

Comment:
This is probably okay as it is and I've probably done things that way in the past. With pydantic BaseSettings, there is a way to define a sequence of resolvers including a config file inside a customize_sources method. I do that here: https://github.com/AllenNeuralDynamics/aind-codeocean-api/blob/377c475d9084bf9802a2e4effc8063a132e48ce9/src/aind_codeocean_api/credentials.py#L136